### PR TITLE
Ronan

### PR DESCRIPTION
**Pull request by RonanXin**
Time serial number in excel format, contamination.txt include time, g/L, num of particle per sample, histogram per sample

### CODE REVIEW

File: particle_counter/main/hmi.m [hunk 1]
```diff
@@ -40,10 +40,13 @@
...
-    
-
-
+    try
+        gui_mainfcn(gui_State, varargin{:});
+    catch ERROR
+        logError = fopen('ERROR.txt','a');
+        fprintf(logError,['%-' int2str(length(ERROR.message)) 's\r\n'],ERROR.message);
+        fclose(logError);
+    end
```
Comment (byocp) [inline particle_counter/main/hmi.m:49]: Very good change.

File: particle_counter/main/hmi.m [hunk 2]
```diff
@@ -375,13 +384,21 @@ function pushbutton1_Callback(hObject, eventdata, handles)
...
 %         fprintf(fileID,'%6.2f %12.8f\r\n', contamination_hist);
-        fprintf(fileID,'%8f %12.8f\r\n', contamination_hist);
+        fprintf(fileID,'%8f, %12.4f, %4d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d, %3d\r\n', contamination_hist(:, Flag.Time));
         fclose(fileID);
-    end
-
 
+    end
+   
+
```
Comment (byocp) [inline particle_counter/main/hmi.m]: Delete the unnecessary empty lines.

File: particle_counter/main/hmi.m [hunk 3]
```diff
@@ -352,22 +355,28 @@ function pushbutton1_Callback(hObject, eventdata, handles)
...
         calculation_noimage = rmfield(calculation,{'raw_image','cropped_raw_image'});
         save(['dataset_evaluation/', TodayDate, '/', filename], 'calculation_noimage');
         imwrite(calculation.raw_image,         cmap, ['dataset_evaluation/', TodayDate, '/', num2str(Flag.Time), '.jpg'], 'jpeg');
         imwrite(calculation.cropped_raw_image * 256, cmap, ['dataset_evaluation/', TodayDate, '/', num2str(Flag.Time), 'C.jpg'], 'jpeg');
-        fileID = fopen(['dataset_evaluation/', TodayDate, '/', 'contamination.txt'],'w');
-        fprintf(fileID,'%8f %12.8f\r\n', contamination_hist);
+ 
+        saveScatter = getframe(handles.axes7);
+        saveScatter = frame2im(saveScatter);
+        imwrite(saveScatter,['dataset_evaluation/', TodayDate, '/', num2str(Flag.Time), 'S.jpg'], 'jpeg');
```
Comment (byocp) [inline particle_counter/main/hmi.m]: What are these three lines about?

File: particle_counter/main/hmi.m [hunk 4]
```diff
@@ -352,22 +355,28 @@ function pushbutton1_Callback(hObject, eventdata, handles)
     if ~exist(['dataset_evaluation\', TodayDate],'dir')
         mkdir('dataset_evaluation\', TodayDate)
     end
+     strSystemTime = m2xdate(now);
+%      contamination_hist(:, Flag.Time) = [strSystemTime, calculation.contam_level_gram_per_liter, calculation.contam_level_num_per_sample];
+    histogram = []; 
+    for k = 1:20
+         histogram = [histogram sum(calculation.equiv_diam<k/10 & calculation.equiv_diam>(k-1)/10)];
```
Comment (byocp) [inline particle_counter/main/hmi.m]: `sum(calculation.equiv_diam < k/  10 & calculation.equiv_diam > (k - 1) / 10)`

Plus, currently `major_axis` is set for diameter calculation in `runtime.m`. Do you want to add this capability here as well? I mean adding capability of having different types of diameter calculation here as well.

**Comment by byocp** [comment]
@RonanXin please also make sure that to choose a proper and descriptive subject for the PR.

File: particle_counter/main/hmi.m (hunk 3 above)
Comment (RonanXin) [inline particle_counter/main/hmi.m]: It's for saving the cropped image with Scatter points in to file

Comment (byocp) [inline particle_counter/main/hmi.m]: It was saving before as well. What changes you have made?

Comment (RonanXin) [inline particle_counter/main/hmi.m]: It was only saving the cropped images, the scatters representing particle were not saved.

Comment (byocp) [inline particle_counter/main/hmi.m]: Got it. Good change.

File: particle_counter/main/hmi.m (hunk 4 above)
Comment (RonanXin) [inline particle_counter/main/hmi.m]: I put it here because I don't know if you want us to change the main.m outputs.

Comment (byocp) [inline particle_counter/main/hmi.m]: My point is at the current revision the histogram results would be always based on the `equiv_diam`, right?

The following link may be helpful: 

http://www.mathworks.com/help/matlab/ref/setfield.html

Comment (RonanXin) [inline particle_counter/main/hmi.m]: From what payman wants, we need a histogram with 0.1mm bins from 0 to 2.0mm in diameter range as an output.

Comment (byocp) [inline particle_counter/main/hmi.m]: "diameter range" can be major_axis, minor_axis, or equiv_diam. agreed?

Comment (RonanXin) [inline particle_counter/main/hmi.m]: yes, you are right.

Comment (byocp) [inline particle_counter/main/hmi.m]: so in that case, please revise code accordingly. thanks.

Comment (RonanXin) [inline particle_counter/main/hmi.m]: but we didn't output the result to file

Comment (byocp) [inline particle_counter/main/hmi.m]: sure. we can chat later about this.

**Comment by byocp** [comment]
LGTM

**Pull request merged**